### PR TITLE
Removing dataplane_rhev_adoption role

### PR DESCRIPTION
**Pull request by fao89**
dataplane_adoption should support RHEV: https://github.com/openstack-k8s-operators/data-plane-adoption/pull/213
`make test-minimal` should work on RHEV

### CODE REVIEW

**Comment by fao89** [comment]
recheck

File: tests/roles/dataplane_adoption/tasks/main.yaml
Comment (fao89) [inline tests/roles/dataplane_adoption/tasks/main.yaml:78]: @archana203 I've added it here

**Comment by fao89** [comment]
recheck

File: tests/roles/dataplane_adoption/tasks/main.yaml
Comment (archana203) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: I still see permission denied with this:
[stack@undercloud-0 ~]$ ssh -i ~/.ssh/id_rsa stack@192.168.23.10 "sudo echo test > /var/lib/nova/compute_id"
bash: line 1: /var/lib/nova/compute_id: Permission denied

Comment (fao89) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: @bogdando @gibizer @SeanMooney do you have any clue on how to make this work?
In our CI we use `root` so it works fine, but on RHEV we have the `stack` user and we are hitting this error

File: tests/roles/dataplane_adoption/tasks/main.yaml
Comment (fao89) [inline tests/roles/dataplane_adoption/tasks/main.yaml:81]: @archana203 could you check if this PR works on RHEV when we skip this step?

File: tests/roles/dataplane_adoption/tasks/main.yaml
Comment (bogdando) [reply tests/roles/dataplane_adoption/tasks/main.yaml]: As @sean-k-mooney have recommended for a similar case, consider using `edpm_user` role to create an `openstack` user (or stack?), then make sure when a task uses ``become:true`` (instead of sudo) it sets the user and group to ansible user, like this https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_nova/tasks/configure.yml#L3-L19

File: tests/roles/dataplane_adoption/tasks/main.yaml
Comment (archana203) [reply tests/roles/dataplane_adoption/tasks/main.yaml:81]: @fao89 Yeah, it worked without this step.

File: tests/roles/dataplane_adoption/tasks/main.yaml
Comment (fao89) [reply tests/roles/dataplane_adoption/tasks/main.yaml]: something like this? https://github.com/openstack-k8s-operators/edpm-ansible/pull/534/files

**Comment by fao89** [comment]
recheck

File: tests/roles/dataplane_adoption/tasks/main.yaml
Comment (gibizer) [inline tests/roles/dataplane_adoption/tasks/main.yaml:78]: if this keeps read access for the nova-compute service to this file then I'm fine with it.

Comment (archana203) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: compute_id is registered with this command. But the id does not match with the one registered in compute service list and computes state is down.
https://privatebin.corp.redhat.com/?2a67ecedadf5a3a2#3qmFnemnvWxERgykPRhiE3FQ3JjR96hiTRzftm49ohSh

Comment (fao89) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: @bogdando @gibizer any idea why the compute state is down?

Comment (gibizer) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: @archana203 could you point me to the job logs?

Comment (gibizer) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: Without logs or troubleshooting I can only guess.
A) the nova-compute service cannot read the compute_id file as it is now written via sudo
B) for some reason the openstack client is different in this case and instead of defaulting to the maximum nova api microversion it defaults to the minimum. If this is the case then you can add `--os-compute-api-version 2.53` to the openstack hypervisor command to force the newer microversion. (Before 2.53 the hypervisor show CLI returned the 'id' not the 'uuid' of the compute node object)

Comment (gibizer) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: I looked at the deployment provided by @archana203 . The OSP18 deployment failed at network config
```
TASK [osp.edpm.edpm_users : Ensure user is present on the host [ openvswitch ]] ***
Monday 22 January 2024  18:18:17 +0000 (0:00:00.630)       0:00:03.397 ********
fatal: [compute-0]: FAILED! => {"changed": false, "msg": "usermod: user openvswitch is currently used by process 15864\n", "name": "openvswitch", "rc": 8}
fatal: [compute-1]: FAILED! => {"changed": false, "msg": "usermod: user openvswitch is currently used by process 15950\n", "name": "openvswitch", "rc": 8}
```
So the new nova_compute container is not yet deployed.

The compute_id is generate properly. It has o+r rights so nova-compute will be able to read it.

Comment (bogdando) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: there was a fix https://github.com/openstack-k8s-operators/edpm-ansible/pull/540

Comment (fao89) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: does it mean sudo isn't needed anymore?

Comment (SeanMooney) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: no sudo is still required

but the issue gibi noted is fixed.

it was unrelated to this change

Comment (SeanMooney) [reply tests/roles/dataplane_adoption/tasks/main.yaml:78]: /var/lib/nova/compute_id is chown'd by kolla when the nova_compute container starts so this will become owned by nova at that point so yes nova will have read access to it.

**Comment by fao89** [comment]
recheck

**Comment by jistr** [comment]
LGTM but i think we need an ack from @archana203 before merging.

**Comment by archana203** [comment]
LGTM, The changes work on RHEV setup without an issue.

**Review by jistr — APPROVED** [review]

**Comment by jistr** [comment]
One thing tho -- the PR has been here for a while and the branch is now 10 commits behind main. We should probably rebase and let the CI run again before merging?

**Comment by fao89** [comment]
just rebased it!

File: tests/roles/dataplane_adoption/vars/rhev.yaml
Comment (fao89) [inline tests/roles/dataplane_adoption/vars/rhev.yaml:78]: 2nd node

**Pull request merged**
